### PR TITLE
gpui: Fix `hide`, `activate` method on Windows to hide/show application

### PR DESCRIPTION
**Pull request by huacnlee**
Release Notes:

- N/A

Continue #18161 to fix `cx.hide`, `cx.activate` method on Windows to hide/show application.

## After

https://github.com/user-attachments/assets/fe0070f9-7844-4c2a-b859-3e22ee4b8d22

### CODE REVIEW

File: crates/gpui/src/platform/windows/platform.rs [hunk 1]
```diff
@@ -314,12 +317,25 @@ impl Platform for WindowsPlatform {
         }
     }
 
-    // todo(windows)
-    fn activate(&self, _ignoring_other_apps: bool) {}
+    fn activate(&self, _ignoring_other_apps: bool) {
+        let mut state = self.hidden_windows.write();
+        state.iter().for_each(|handle| unsafe {
+            ShowWindow(*handle, SW_SHOW).ok().log_err();
```
Comment (huacnlee) [inline crates/gpui/src/platform/windows/platform.rs:323]: Here maybe need to improve to support `_ignoring_other_apps` option, I will try other option for `ShowWindow` method.

https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-showwindow

Comment (mikayla-maki) [reply crates/gpui/src/platform/windows/platform.rs]: I think that is a [macOS specific](https://developer.apple.com/documentation/appkit/nsapplication/1428468-activateignoringotherapps) behavior, so it should be ok to drop this.

**Review by mikayla-maki — CHANGES_REQUESTED** [review]
This PR should wait until https://github.com/zed-industries/zed/pull/18161 is merged, then we can rebase and merge this one :D

**Comment by huacnlee** [comment]
This is ready to merge unlit #18161 merged, it has #18161 difference code

**Review by mikayla-maki — APPROVED** [review]

**Pull request merged**